symphony-bdk-spring/symphony-bdk-core-spring-boot-starter/src/main/java/com/symphony/bdk/spring/annotation/SlashAnnotationProcessor.java

```diff
@@ -0,0 +1,155 @@
+package com.symphony.bdk.spring.annotation;
+
+import static com.symphony.bdk.core.activity.command.SlashCommand.slash;
+
+import com.symphony.bdk.core.activity.ActivityRegistry;
+import com.symphony.bdk.core.activity.command.CommandContext;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.aop.framework.autoproxy.AutoProxyUtils;
+import org.springframework.aop.scope.ScopedObject;
+import org.springframework.aop.scope.ScopedProxyUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanInitializationException;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.core.MethodIntrospector;
+import org.springframework.core.annotation.AnnotatedElementUtils;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ClassUtils;
+import org.springframework.util.CollectionUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+/**
+ *
+ * @see <a href="https://docs.spring.io/spring-framework/docs/current/reference/html/core.html#beans-factory-extension-bpp">BeanPostProcessor</a>
+ * @see <a href="https://github.com/spring-projects/spring-framework/blob/master/spring-context/src/main/java/org/springframework/context/event/EventListenerMethodProcessor.java">EventListenerMethodProcessor.java</a>
+ */
+@Slf4j
+public class SlashAnnotationProcessor implements BeanPostProcessor, ApplicationContextAware {
+
+  /** The {@link ActivityRegistry} is used here to register the slash activities */
+  private final ActivityRegistry registry;
+
+  private ConfigurableApplicationContext applicationContext;
+
+  public SlashAnnotationProcessor(ActivityRegistry registry) {
+    this.registry = registry;
+  }
+
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    this.applicationContext = (ConfigurableApplicationContext) applicationContext;
+  }
+
+  @Override
+  public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+
+    if (!ScopedProxyUtils.isScopedTarget(beanName)) {
+      Class<?> type = null;
+      try {
+        type = AutoProxyUtils.determineTargetClass(this.applicationContext.getBeanFactory(), beanName);
+      }
+      catch (Throwable ex) {
+        // An unresolvable bean type, probably from a lazy bean - let's ignore it.
```

Comment:
is it the reason? I thought here the bean is already instanciated? can't we use bean.getClass() to retrieve it by the way?